lib/web_ui/lib/src/engine/text/fragmenter.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:ui/ui.dart' as ui;
+
+import 'canvas_paragraph.dart';
+import 'line_breaker.dart';
+import 'text_direction.dart';
+
+abstract class TextFragmenter {
+  const TextFragmenter(this.paragraph);
+
+  final CanvasParagraph paragraph;
```

Comment:
I imagine the goal is to have a CanvasKit implementation of this class. If so, we should probably not have a dependency on `CanvasParagraph` here?

Reply:
Done.